### PR TITLE
DDF-4439 Upgrade jackson databind to version 2.9.8

### PR DESCRIPTION
**Pull request by bellcc**
#### What does this PR do?
- Upgrades jackson to version 2.9.8

#### Who is reviewing it? 

@Corey-Collins 
@garrettfreibott 

#### Select relevant component teams: 
@codice/build 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@tbatie 

#### How should this be tested?

- Version upgrade so full build is sufficient

#### What are the relevant tickets?
[DDF-4439](https://codice.atlassian.net/browse/DDF-4439)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bellcc** [comment]
build now

File: platform/solr/solr-dependencies/pom.xml [hunk 1]
```diff
@@ -168,7 +168,7 @@
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
-            <version>2.9.5</version>
+            <version>${jackson.version}</version>
```
Comment (Bdthomson) [inline platform/solr/solr-dependencies/pom.xml]: We weren't using this property before?

Comment (bellcc) [reply platform/solr/solr-dependencies/pom.xml]: It's being used in one other place in DDF.

**Comment by bellcc** [comment]
build now

**Comment by bellcc** [comment]
We have two versions of jackson defined. The CVEs that this latest build failed on are supposedly fixed in version 2.9.7.

Going to sync them both at 2.9.7 to see if it works first. Then I'll try 2.9.8

**Comment by bellcc** [comment]
build now

**Comment by bellcc** [comment]
All of the CVEs that I am suppressing affect versions 2.9.7 and lower

**Comment by bellcc** [comment]
build now

**Comment by bellcc** [comment]
build now

**Comment by bellcc** [comment]
build now

**Comment by bellcc** [comment]
build now

**Comment by bellcc** [comment]
build now

**Comment by paouelle** [comment]
@clockard - It would seem this is also affecting the 2.13.x release and should also be committed there.

**Comment by clockard** [comment]
@paouelle it is probably worth while to port this. @brjeter what are your thoughts?

**Comment by paouelle** [comment]
Builds on the release branch are not going to pass until we do 😞

**Comment by brjeter** [comment]
@clockard: I agree that this should be ported to 2.13.x

**Comment by bellcc** [comment]
I'll backport this to 2.13.x right now

**Comment by paouelle** [comment]
Let's make sure we get an answer on the above question from @pklinef. I know we have hardcoded versions of libraries for Solr for good reasons. I just want to make sure that it is safe to upgrade the Jackson library used by Solr as done here.

**Comment by bellcc** [comment]
@paouelle @brjeter @clockard Builds aren't going to work on the release branch unless I suppress another CVE as well. 

`org.apache.karaf.jaas.modules-4.1.2.jar: CVE-2018-11786`

**Comment by brjeter** [comment]
@bellcc good find. That sounds good to me. We are on Karaf 4.2.1 in 2.13.x, so this is just another case where the `maven-dependency-plugin` is triggering on a CVE that doesn't actually apply.

File: platform/solr/solr-dependencies/pom.xml [hunk 2]
```diff
@@ -158,22 +158,22 @@
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-annotations</artifactId>
-            <version>2.9.5</version>
+            <version>${jackson.version}</version>
```
[the remark this answers is not in the document]
Comment (pklinef) [reply platform/solr/solr-dependencies/pom.xml]: In general we try to use the exact same version Solr uses.  But we have used newer versions in Karaf, Solr, and other projects for reasons like OWASP.  If that applies then a comment should be added for why we are not using the version used by Solr.

[the remark this answers is not in the document]
Comment (bellcc) [reply platform/solr/solr-dependencies/pom.xml]: With this change DDF and DDF's Solr are going to use the same version so no comment will be necessary.

[the remark this answers is not in the document]
Comment (paouelle) [reply platform/solr/solr-dependencies/pom.xml]: I think the request was to comment as to the fact that you were no longer using the same version of Jackson as the Solr project (not ours) uses.

[the remark this answers is not in the document]
Comment (bellcc) [reply platform/solr/solr-dependencies/pom.xml]: Comment added!

**Comment by bellcc** [comment]
build now

**Pull request merged**